uwblueprint/focus-on-nature
# Camp Form Endpoints

## Notion ticket link

[Ticket Name](https://uwblueprintexecs.notion.site/Camp-Form-Endpoints-2ba91c35ffae4f25b72710c47862a12b)



## Implementation description
* POST endpoint to create form questions `/camp/{campId}/form/`
* PUT endpoint to edit form questions by id ` /camp/{campId}/form/{formQuestionId}`
* PATCH endpoint to append new form questions `/camp/{campId}/form`
* DELETE endpoint to delete a form question by id `/camp/{campId}/form/{formQuestionId}`



## Steps to test
1. Ensure service starts successfully
2. Try and push test data to the endpoints listed in the summary. Examples of test data listed below
3. Ensure that 200 level responses are returned and that MongoDB Atlas contains the correct data
<br />
* POST
[image](https://user-images.githubusercontent.com/64621569/172068346-aa9d1a78-511d-4e87-8b0f-c8c7622275e1.png)
<br />
* PUT
[image](https://user-images.githubusercontent.com/64621569/172068369-b7294c27-1f5d-4405-a909-24a47dd02644.png)
<br />
* PATCH
[image](https://user-images.githubusercontent.com/64621569/172068422-7050cdff-9165-404c-83a5-ec5dc433a679.png)
(not sure why pics are not appearing, just copy-paste link or sum)



## What should reviewers focus on?
* Ensure code reusability and code handles all edge cases
* Should PATCH and POST requests just return the ids of the form questions, or the actual form questions themselves


## Checklist
